src/app/services/tests/ChatService.spec.ts

```diff
@@ -42,61 +42,72 @@ describe('ChatService', () => {
+            // Then it logged an error
+            const errorMessage: string = 'Already observing same chat';
+            const errorData: JSONValue = { chatId: 'id' };
+            expect(ErrorLoggerService.logError).toHaveBeenCalledWith('ChatService', errorMessage, errorData);
         }));
         it('should throw when observing a second chat while a first one is already being observed', fakeAsync(async() => {
-            await chatDAO.set('id', EMPTY_CHAT);
-            // given a chat that is observed
-            service.startObserving('id', (_: MGPOptional<Chat>) => { });
-            // when trying to observe another chat, then an error is thrown
-            expect(() => service.startObserving('id2', (_: MGPOptional<Chat>) => { })).toThrowError(`Cannot ask to watch 'id2' while watching 'id'`);
+            spyOn(ErrorLoggerService, 'logError').and.callFake(ErrorLoggerServiceMock.logError);
+            // Given a chat that is observed
+            await service.createNewChat('id');
+            service.startObserving('id', new FirebaseCollectionObserver<Message>(() => {}, () => {}, () => {}));
+            // When trying to observe another chat, then an error is thrown
+            service.startObserving('id2', new FirebaseCollectionObserver<Message>(() => {}, () => {}, () => {}));
+            // Then it logged an error
+            const errorMessage: string = 'Already observing another chat';
+            const errorData: JSONValue = { chatId: 'id2', followedChatId: 'id' };
+            expect(ErrorLoggerService.logError).toHaveBeenCalledWith('ChatService', errorMessage, errorData);
         }));
         it('should stop following updates after stopObserving is called', fakeAsync(async() => {
-            let resolvePromise: (chat: Chat) => void;
-            const promise: Promise<Chat> = new Promise((resolve: (chat: Chat) => void) => {
+            let resolvePromise: (messages: Message[]) => void;
+            const promise: Promise<Message[]> = new Promise((resolve: (messages: Message[]) => void) => {
                 resolvePromise = resolve;
             });
-            const callback: (chat: MGPOptional<Chat>) => void = (chat: MGPOptional<Chat>) => {
-                if (chat.isPresent() && chat.get().messages.length > 0) {
-                    resolvePromise(chat.get());
+            const callback: (messages: MessageDocument[]) => void = (messages: MessageDocument[]) => {
+                if (messages.length > 0) {
+                    resolvePromise(messages.map((doc: MessageDocument) => doc.data));
                 }
             };
             expect(service.isObserving()).toBe(false);
-            await chatDAO.set('id', EMPTY_CHAT);
+            await service.createNewChat('id');
 
             // given a chat that is not observed anymore
-            service.startObserving('id', callback);
+            service.startObserving('id', new FirebaseCollectionObserver<Message>(callback, () => {}, () => {}));
```

Comment:
kind of a mismatch between given's comment and given's code 📟

Reply:
updated given